### DISCUSSION TITLE
What should go in the `@dgrants/utils` and `@dgrants/types` packages?

### DISCUSSION THREAD

**Discussion by mds1** [dcgtc/dgrants#114] [Dev Meta] [answered]
Context that inspired this discussion: https://github.com/dcgtc/dgrants/pull/111#discussion_r690182355

Right now the `dgrants` monorepo contains the following packages:

- `app`
- `contracts`
- `doc`
- `types`
- `utils`

Where `app` and `contracts` are self-explanatory, and`doc` is a WIP package for documentation.

Then we have `types` and `utils` which currently have unclear purposes, and I struggle figuring out when to put things into those packages. I see three possible approaches:

1. Only add something to the `utils` and `types` packages if they _are_ currently shared between other packages. 
2. Add something to the `utils` and `types` package if it might be shared eventually between other packages eventually
3. Add something to the `utils` and `types` package if it might be helpful to someone building their own frontend

Currently I've been assuming 1, but I think this is something we should agree upon and add to the CONTRIBUTING.md file

**Comment by thelostone-mc** [comment]
- Would we want to have a separate doc folder / just have readme within each package? (that way within each package you find the needed documentation ). I'm good with either
- Thoughts on renaming utils to common (checkpoint below on why)
- Do we need a separate folder for types? Could it be inside `common` package ? 

> Then we have types and utils which currently have unclear purposes, and I struggle to figure out when to put things into those packages. I see three possible approaches:

Hmm, I'd lean towards 2 or 1 for now and maybe we revisit this later?

**Comment by apbendi** [comment, marked as answer]
I wonder if we eventually might want to develop a `dgrants` SDK, i.e. an npm we publish that makes it easy to integrate with dgrants from JavaScript on either the backend or frontend? Perhaps that's a separate discussion.

Unless/until we decide to go that route, I'd opt for "lazy extraction," i.e. option 1: only pulling something out once it needs to be shared.
